src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -3,150 +3,157 @@
+
+            await itemLoader.InitializeAsync();
+
+            return itemLoader;
         }
 
-        public PackageItemLoader(
+        // For unit testing purposes
+        internal static async ValueTask<PackageItemLoader> CreateAsync(
+            IServiceBroker serviceBroker,
             PackageLoadContext context,
-            IPackageFeed packageFeed,
+            IReadOnlyCollection<PackageSourceContextInfo> packageSources,
+            ContractItemFilter itemFilter,
+            INuGetSearchService searchService,
             string searchText = null,
             bool includePrerelease = true,
-            IPackageFeed recommenderPackageFeed = null)
+            bool useRecommender = true)
+        {
+            var itemLoader = new PackageItemLoader(
+                serviceBroker,
+                context,
+                packageSources,
+                itemFilter,
+                searchText,
+                includePrerelease,
+                useRecommender);
+
+            await itemLoader.InitializeAsync(searchService);
+
+            return itemLoader;
+        }
+
+        private async ValueTask InitializeAsync(INuGetSearchService searchService = null)
         {
-            if (context == null)
+            _searchFilter = new SearchFilter(includePrerelease: _includePrerelease)
             {
-                throw new ArgumentNullException(nameof(context));
-            }
-            _context = context;
+                SupportedFrameworks = await _context.GetSupportedFrameworksAsync()
+            };
 
-            if (packageFeed == null)
+            _searchService = searchService ?? await GetSearchServiceAsync(CancellationToken.None);
+            _serviceBroker.AvailabilityChanged += OnAvailabilityChanged;
+        }
+
+        private void OnAvailabilityChanged(object sender, BrokeredServicesChangedEventArgs e)
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
-                throw new ArgumentNullException(nameof(packageFeed));
-            }
-            _packageFeed = packageFeed;
-            _recommenderPackageFeed = recommenderPackageFeed;
+                _searchService?.Dispose();
+                _searchService = await GetSearchServiceAsync(CancellationToken.None);
+            }).PostOnFailure(nameof(PackageItemLoader), nameof(OnAvailabilityChanged));
+        }
 
-            _searchText = searchText ?? string.Empty;
-            _includePrerelease = includePrerelease;
+        private async ValueTask<INuGetSearchService> GetSearchServiceAsync(CancellationToken cancellationToken)
+        {
+#pragma warning disable ISB001 // Dispose of proxies
+            INuGetSearchService searchService = await _serviceBroker.GetProxyAsync<INuGetSearchService>(NuGetServices.SearchService, cancellationToken);
+#pragma warning restore ISB001 // Dispose of proxies
+            Assumes.NotNull(searchService);
+            return searchService;
```

Comment:
i believe this analyzer is doing what it should. we need to be careful when we keep these proxies around...as we are doing in this case.

Reply:
When a disposable object is being returned from a method, it should not be disposed by the method returning it, otherwise the caller will receive an object that is already disposed. The analyzer should support this.